src/main/webapp/components/workspace/workspace.js

```diff
@@ -160,9 +171,49 @@ export default () => {
             overflow: 'auto',
           }}
         >
-          <Typography variant="h4" component="h1" style={{ padding: 20 }}>
-            {title}
-          </Typography>
+          <div
+            style={{
+              display: 'flex',
+              alignItems: 'baseline',
+            }}
+          >
+            <TextField
+              variant="outlined"
+              style={{ margin: '10px', width: '70%' }}
+              defaultValue={title}
+              onBlur={e => {
+                saveWorkspace(setSaving, { title: e.target.value, queries })
```

Comment:
Might be good to have a function that returns the current workspace merged with updated properties you pass the function, instead of getting all the need properties manually

Reply:
IIRC the graphql resolver supports partial updates. I think the only reason you need to pass in queries, is because it defaults to an empty array in the hook: https://github.com/connexta/revelio/blob/master/src/main/webapp/components/workspace/hooks/use-save-workspace.tsx#L21-L23

Reply:
I tried doing the saveMetacard mutation above with just the title, and it removed all the queries on the workspace :crying_cat_face:

Reply:
@samuelechu the reason that is, is because `use-save-workspace` has the following code
```const queries = (workspace.queries || []).map((query: QueryType) => ({
      id: query.id,
}))```
so if no queries are passed in it defaults to an empty array. Hence the removal of all queries in the workspace if you pass just the title.

Reply:
Here's the query I did in graphiql (`saveQueryToWorkspace`). 
![image](https://user-images.githubusercontent.com/4467636/79791169-b8fcc280-8301-11ea-92da-bb8ff6252527.png)

Would you verify that you're able to do a partial update? It didn't happen for me.

Reply:
if I removed queries from `workspaceAttrs`, all queries were removed from the workspace. When I removed title however, the current title was preserved.

Reply:
> @samuelechu the reason that is, is because `use-save-workspace` has the following code
> 
> ```
> const queries = (workspace.queries || []).map((query: QueryType) => ({
>       id: query.id,
> }))
> ```
> 
> so if no queries are passed in it defaults to an empty array. Hence the removal of all queries in the workspace if you pass just the title.

My mistake, I didn't clarify that I was doing a mutation in graphiql

Reply:
I think this is a bug with our graphql, when it sends the save request, queries is a function: 
![image](https://user-images.githubusercontent.com/10997562/79876275-10556e00-83a8-11ea-88bb-a31c3524e895.png)